streams-bootstrap/src/main/java/com/bakdata/kafka/KafkaStreamsApplication.java

```diff
@@ -271,13 +267,31 @@ protected Properties createKafkaProperties() {
         kafkaConfig.setProperty(StreamsConfig.producerPrefix(ProducerConfig.COMPRESSION_TYPE_CONFIG), "gzip");
 
         // topology
-        kafkaConfig.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, this.getUniqueAppId());
+        kafkaConfig.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, this.getApplicationId());
 
         this.configureDefaultSerde(kafkaConfig);
         kafkaConfig.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, this.getBrokers());
         return kafkaConfig;
     }
 
+    private String getApplicationId() {
+        final String uniqueAppId = this.getUniqueAppId();
+        if (uniqueAppId == null) {
+            if (this.uniqueAppId == null) {
+                throw new IllegalArgumentException("Must pass --application-id or implement #getUniqueAppId()");
```

Comment:
You removed the abstract method. How can the user implement the `getUniqueAppId()`?

Reply:
The field is covered by Lombok `@Getter`, tests still override it

Reply:
We can also change the field name and do not generate a Getter for it 🤔

Reply:
It was just slightly confusing that the flag was --application-id and the field name was `uniqueAppId`